### PR TITLE
#110 item displaying

### PR DESCRIPTION
**Pull request by Zeavee**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/artificial_intelligence/Movement.java [hunk 1]
```diff
@@ -1,10 +1,65 @@
...
+    private double orientation;
+    private MovementType movementType;
+    private double sinusAmplitude = 1;
+    private double sinusAngle;
+
+    public Movement() {
+        acceleration = 0;
+        velocity = 0;
+        orientation = 0;
+        movementType = MovementType.LINEAR;
```
Comment (kangpl) [inline app/src/main/java/ch/epfl/sdp/artificial_intelligence/Movement.java:15]: Shall we also initialize sinusAngle?

Comment (Zeavee) [reply app/src/main/java/ch/epfl/sdp/artificial_intelligence/Movement.java:15]: We should yes, in this branch I only moved this code to another class, so I did not change anything. Normally it should already be initialized to 0, but we can do it explicitely.

File: app/src/main/java/ch/epfl/sdp/entity/Inventory.java [hunk 2]
```diff
@@ -78,12 +78,18 @@ private void useHealthPack(Healthpack healthpack) {
     }
 
     public void addItem(String itemName) {
+        if (!items.containsKey(itemName)) {
+            items.put(itemName, 0);
+        }
```
Comment (kangpl) [inline app/src/main/java/ch/epfl/sdp/entity/Inventory.java:83]: Good to check

File: app/src/main/java/ch/epfl/sdp/map/MapsActivity.java [hunk 3]
```diff
@@ -12,11 +12,18 @@
...
+import ch.epfl.sdp.item.Healthpack;
+import ch.epfl.sdp.item.Item;
 import ch.epfl.sdp.item.Scan;
 
 public class MapsActivity extends FragmentActivity implements OnMapReadyCallback {
     public static final MapApi mapApi = new GoogleMapApi();
 
+    public static MapApi getMapApi() {
+        return mapApi;
+    }
```
Comment (kangpl) [inline app/src/main/java/ch/epfl/sdp/map/MapsActivity.java:25]: Since mapApi is already public, do we need this getter function?

Comment (Zeavee) [reply app/src/main/java/ch/epfl/sdp/map/MapsActivity.java:25]: I tried to use this to mock the static mapapi, but I will remove this (it came back after swapping between commits to check leaderboard). Thank you for telling me :)

File: app/src/main/java/ch/epfl/sdp/artificial_intelligence/Movement.java [hunk 4]
```diff
@@ -1,10 +1,67 @@
...
+
+public class Movement {
+    private float velocity;
+    private float acceleration;
+    private double orientation;
+    private MovementType movementType;
+    private double sinusAmplitude;
+    private double sinusAngle;
+
+    public Movement() {
```
Comment (RaphaelBonatti) [inline app/src/main/java/ch/epfl/sdp/artificial_intelligence/Movement.java:11]: That's a useful code refactoring there, thanks.

File: app/src/main/java/ch/epfl/sdp/artificial_intelligence/MovementType.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package ch.epfl.sdp.artificial_intelligence;
+
+public enum MovementType {
```
Comment (RaphaelBonatti) [inline app/src/main/java/ch/epfl/sdp/artificial_intelligence/MovementType.java:3]: Good choice of name.

File: app/src/main/java/ch/epfl/sdp/artificial_intelligence/MovingArtificialEntity.java [hunk 6]
```diff
@@ -7,25 +7,19 @@
...
-    private float velocity;
-    private float acceleration;
-    private double orientation;
-    private Movement movement;
     private boolean moving;
     private Boundable bounds;
-    public double sinusAmplitude = 1;
-    public double sinusAngle;
+    private Movement movement;
+    private boolean isActive;
```
Comment (RaphaelBonatti) [inline app/src/main/java/ch/epfl/sdp/artificial_intelligence/MovingArtificialEntity.java:13]: I would rather call it isMoving, what do you think?

Comment (Zeavee) [reply app/src/main/java/ch/epfl/sdp/artificial_intelligence/MovingArtificialEntity.java:13]: This symbolize the fact that the entity is still displayed on the map or not (it means that if this Boolean is set to false, we undisplay the entity and we remove it from the displayable list ). So isMoving is also a bad choice, but yeah, we would need to change this

**Pull request merged**